packages/strapi-mongoose/lib/relations.js

```diff
@@ -0,0 +1,353 @@
+'use strict';
+
+/**
+ * Module dependencies
+ */
+
+// Public node modules.
+const _ = require('lodash');
+
+module.exports = {
+  getModel: function (model, plugin) {
+    return _.get(strapi.plugins, [plugin, 'models', model]) || _.get(strapi, ['models', model]) || undefined;
+  },
+
+  update: async function (params) {
+    const virtualFields = [];
+    const response = await this
+      .findOne({
+        [this.primaryKey]: params[this.primaryKey] || params.id
+      })
+      .populate(this.associations.map(x => x.alias).join(' '));
+
+    // Only update fields which are on this document.
+    const values = params.parseRelationships === false ? params.values : Object.keys(JSON.parse(JSON.stringify(params.values))).reduce((acc, current) => {
+      const association = this.associations.filter(x => x.alias === current)[0];
+      const details = this._attributes[current];
+
+      if (_.get(this._attributes, `${current}.isVirtual`) !== true && _.isUndefined(association)) {
+        acc[current] = params.values[current];
+      } else {
+        switch (association.nature) {
+          case 'oneWay':
+            acc[current] = _.get(params.values[current], this.primaryKey, params.values[current]) || null;
+
+            break;
+          case 'oneToOne':
+            if (response[current] !== params.values[current]) {
+              const value = _.isNull(params.values[current]) ? response[current] : params.values;
```

Comment:
No, it will set the constant variable `value` to `params.values[current]` whereas it should be set `params.values`. Take another look at the condition ;)